extensions/machine-learning-services/src/modelManagement/deployedModelService.ts

```diff
@@ -207,20 +194,24 @@ export class DeployedModelService {
 		`;
 	}
 
-	public getUpdateModelQuery(currentDatabaseName: string, model: RegisteredModel): string {
+	public getInsertModelQuery(currentDatabaseName: string, model: RegisteredModel): string {
 		let updateScript = `
-		UPDATE ${utils.getRegisteredModelsTowPartsName(this._config)}
-		SET
-		name = '${utils.doubleEscapeSingleQuotes(model.title || '')}',
-		version = '${utils.doubleEscapeSingleQuotes(model.version || '')}',
-		description = '${utils.doubleEscapeSingleQuotes(model.description || '')}'
-		WHERE artifact_id = ${model.id}`;
+		Insert into ${utils.getRegisteredModelsTowPartsName(this._config)}
+		(artifact_name, group_path, artifact_content, name, version, description)
+		values (
+			'${utils.doubleEscapeSingleQuotes(model.artifactName || '')}',
+			'ADS',
+			${utils.doubleEscapeSingleQuotes(model.content || '')},
+			'${utils.doubleEscapeSingleQuotes(model.title || '')}',
+			'${utils.doubleEscapeSingleQuotes(model.version || '')}',
+			'${utils.doubleEscapeSingleQuotes(model.description || '')}')
+		`;
 
 		return `
 		${utils.getScriptWithDBChange(currentDatabaseName, this._config.registeredModelDatabaseName, updateScript)}
 		SELECT artifact_id, artifact_name, name, description, version, created
 		FROM ${utils.getRegisteredModelsThreePartsName(this._config)}
-		WHERE artifact_id = ${model.id};
+		WHERE artifact_id = SCOPE_IDENTITY();
```

Comment:
this already defined...? couldnt see here

Reply:
SCOPE_IDENTITY is a system function to get the id of lasted added data in current scope

---
In reply to: [406535923](https://github.com/microsoft/azuredatastudio/pull/9869#discussion_r406535923) [](ancestors = 406535923)